src/topp/CometAdapter.cpp

```diff
@@ -367,68 +370,50 @@ class TOPPCometAdapter :
         binary_group = binary_modifications[var_mod_index];
       }
 
-      //TODO support mod-specific limit (default for now is the overall max per peptide)
-      int max_current_mod_per_peptide = max_variable_mods_in_peptide;
-      //TODO support term-distances?
-      int term_distance = -1;
-      int nc_term = 0;
+      VarMod temp_var_mod(mod, binary_group, max_variable_mods_in_peptide);
 
-      //TODO support agglomeration of Modifications to same AA. Watch out for nc_term value then.
-      if (mod->getTermSpecificity() == ResidueModification::C_TERM)
-      {
-        if (mod->getOrigin() == 'X')
-        {
-          residues = "c";
-        } // else stays mod.getOrigin()
-        term_distance = 0;
-        // Since users need to specify mods that apply to multiple residues/terms separately
-        // 3 and -1 should be equal for now.
-        nc_term = 3;
-      }
-      else if (mod->getTermSpecificity() == ResidueModification::N_TERM)
-      {
-        if (mod->getOrigin() == 'X')
-        {
-          residues = "n";
-        } // else stays mod.getOrigin()
-        term_distance = 0;
-        // Since users need to specify mods that apply to multiple residues/terms separately
-        // 2 and -1 should be equal for now.
-        nc_term = 2;
-      }
-      else if (mod->getTermSpecificity() == ResidueModification::PROTEIN_N_TERM)
+      modifications_given.push_back(temp_var_mod);
+
+    }
+
+    // concatenate residues
+    for (Size i = 0; i < modifications_given.size() - 1; i++)
+    {
+      for (Size j = i+1; j < modifications_given.size(); j++)
       {
-        if (mod->getOrigin() == 'X')
+        if (modifications_given[i].isMergeableWith(modifications_given[j]))
         {
-          residues = "n";
-        } // else stays mod.getOrigin()
-        term_distance = 0;
-        nc_term = 0;
+          modifications_given[i].merge(modifications_given[j]);
+
+          // destruct after merge
+          modifications_given[j].~VarMod();
```

Comment:
Code smell. Doesn't make sense to destruct the value here as it leaves modifications_given[j] in a destructed state.

Reply:
the idea behind destructing merged modifications is to have only unique modifications, redundant will be destroyed this would help majorly in later stages where modifications_given.size() will be only consisting count of unique modifications.

Reply:
Calling the destructor will have no effect on the size of the container.
This code needs some rework as Timo suggested.